### PR TITLE
In 'kubectl describe', find controllers with ControllerRef, instead of showing the original creator

### PR DESCRIPTION
**Pull request by janetkuo**
@enisoc @kargakis @kubernetes/sig-apps-pr-reviews @kubernetes/sig-cli-pr-reviews 

```release-note
In 'kubectl describe', find controllers with ControllerRef, instead of showing the original creator.
```

### CODE REVIEW

**Comment by smarterclayton** [comment]
This is really late for 1.6 - do we really have to do this?

**Comment by janetkuo** [comment]
I don't see this as release blocking, but current `kubectl describe` currently shows the creator as `Controller`, not the actually controller from `ControllerRef`. This may cause user confusion.

File: pkg/printers/internalversion/describe.go [hunk 1]
```diff
@@ -538,7 +539,12 @@ func describePod(pod *api.Pod, events *api.EventList) (string, error) {
 			w.Write(LEVEL_0, "Message:\t%s\n", pod.Status.Message)
 		}
 		w.Write(LEVEL_0, "IP:\t%s\n", pod.Status.PodIP)
-		w.Write(LEVEL_0, "Controllers:\t%s\n", printControllers(pod.Annotations))
+		if createdBy := printCreator(pod.Annotations); len(createdBy) > 0 {
+			w.Write(LEVEL_0, "Created By:\t%s\n", createdBy)
+		}
+		if controlledBy := printController(pod); len(controlledBy) > 0 {
+			w.Write(LEVEL_0, "Controller:\t%s\n", controlledBy)
```
Comment (kargakis) [inline pkg/printers/internalversion/describe.go]: "Controller" is probably one of the most overused words in the community. How about "Controlled By"?

File: pkg/printers/internalversion/describe.go [hunk 2]
```diff
@@ -1338,6 +1351,9 @@ func describeReplicaSet(rs *extensions.ReplicaSet, events *api.EventList, runnin
 		w.Write(LEVEL_0, "Selector:\t%s\n", metav1.FormatLabelSelector(rs.Spec.Selector))
 		printLabelsMultiline(w, "Labels", rs.Labels)
 		printAnnotationsMultiline(w, "Annotations", rs.Annotations)
+		if controlledBy := printController(rs); len(controlledBy) > 0 {
+			w.Write(LEVEL_0, "Controller:\t%s\n", controlledBy)
```
Comment (kargakis) [inline pkg/printers/internalversion/describe.go]: Similar comment as above

**Comment by enisoc** [comment]
@janetkuo Since this is not release-blocking, we should move it to the `v1.6.1` or `v1.7` milestone.

File: pkg/printers/internalversion/describe.go (hunk 2 above)
Comment (janetkuo) [inline pkg/printers/internalversion/describe.go]: Done

File: pkg/printers/internalversion/describe.go (hunk 1 above)
Comment (janetkuo) [inline pkg/printers/internalversion/describe.go]: Done

**Comment by janetkuo** [comment]
@kargakis PTAL

File: pkg/printers/internalversion/describe.go [hunk 3]
```diff
@@ -583,7 +589,14 @@ func describePod(pod *api.Pod, events *api.EventList) (string, error) {
...
 
-func printControllers(annotation map[string]string) string {
+func printController(controllee metav1.Object) string {
+	if controllerRef := controller.GetControllerOf(controllee); controllerRef != nil {
+		return fmt.Sprintf("%s/%s", controllerRef.Kind, controllerRef.Name)
+	}
+	return ""
+}
+
+func printCreator(annotation map[string]string) string {
```
Comment (kargakis) [inline pkg/printers/internalversion/describe.go]: I am wondering if created-by is used for anything else that the controller ref cannot. Can you spawn a separate issue to deprecate it and start removing it from places we can while keeping backwards compatibility?

Comment (janetkuo) [inline pkg/printers/internalversion/describe.go]: Sure

Comment (janetkuo) [inline pkg/printers/internalversion/describe.go]: Filed #44407

**Pull request merged**